File: connectors/connector-mantis/src/main/scala/com/taskadapter/connector/mantis/MantisField.scala
package com.taskadapter.connector.mantis

import com.taskadapter.connector.Field
import com.taskadapter.model._

object MantisField {

  val id = Field("Id")
  val summary = Field("Summary")
  val description = Field("Description")
  val assignee = Field("Assignee")
  val dueDate = Field.date("Due Date")
  val createdOn = Field.date("Created On")
  val updatedOn = Field.date("Updated On")
  val priority = Field.integer("Priority")

  val fields = List(summary, description, assignee, dueDate, createdOn, updatedOn, priority)

  // id field is not in the suggested list because typically
  // id from one system cannot be directly used as id in another system.
  private def suggestedStandardFields = Map(summary -> Summary,
    description -> Description,
    assignee -> Assignee,
    dueDate -> DueDate,
    createdOn -> CreatedOn,
    updatedOn -> UpdatedOn,
    priority -> Priority
  )

  def getSuggestedCombinations(): Map[Field, StandardField] = {
    suggestedStandardFields
  }
}

File: connectors/connector-mantis/src/main/java/com/taskadapter/connector/mantis/MantisField.java
package com.taskadapter.connector.mantis;

import com.taskadapter.model.AllFields;
import com.taskadapter.model.Field;

import java.util.List;

public class MantisField {
    public static List<Field<?>> fields = List.of(
            AllFields.summary,
            AllFields.description,
            AllFields.assigneeFullName,
            AllFields.assigneeLoginName,
            AllFields.dueDate,
            AllFields.createdOn,
            AllFields.updatedOn,
            AllFields.priority);
}
